orttraining/orttraining/training_api/utils.cc

```diff
@@ -63,12 +63,22 @@ Status OrtValueLike(const SessionState& sess_state, const OrtValue& input_val, O
   auto element_type = param_tensor.DataType();
   auto p_tensor = std::make_unique<Tensor>(element_type, shape, allocator);
 
-  // TODO: handle CUDA memset
   if (tensor_location.device.Type() == OrtDevice::CPU ||
       tensor_location.mem_type == OrtMemTypeCPUInput ||
       tensor_location.mem_type == OrtMemTypeCPUOutput) {
     memset(p_tensor->MutableDataRaw(), 0, p_tensor->SizeInBytes());
+  } else if (tensor_location.device.Type() == OrtDevice::GPU) {
+    // Use a tensor on cpu and copy it over to the desired device using
```

Comment:
Is there a perf efficient way to do this? Given this is during initialization it may be OK but wondering whether we can use CudaMemset here?